vm/src/pyobject.rs

```diff
@@ -871,7 +873,7 @@ pub enum PyObjectPayload {
         elements: objdict::DictContentType,
     },
     Set {
-        elements: HashMap<usize, PyObjectRef>,
+        elements: HashMap<BigInt, PyObjectRef>,
```

Comment:
Why change the hash result to `BigInt`? I thought that a hash always produces a native sized unsigned integer like `usize`. Is there a reason for using `BigInt` instead?

Reply:
__hash__ returns a python `int` which maps to `BigInt` in Rust-land :)

Reply:
Yes, this is certainly true, but the range of this BigInt should fall into a usize category, following that most hashes are calculated using rust hashers which return usize. But then again, we could have python hashers which return larger numbers, so I guess you have a valid point!

Reply:
https://github.com/python/cpython/blob/master/Objects/typeobject.c#L6379

if a returned Python `int` is out of range, it gets hashed by builtin `int.__hash__` hashing function. Thus, the result should always fit in an `usize`.

Reply:
Maybe we can take the hash of the bigint and use that? That should be a `usize` thingy.

Reply:
Looks like we could do something like this: https://stackoverflow.com/questions/50484012/can-a-biginteger-be-truncated-to-an-i32-in-rust
I'm fairly busy for the next few days - but there's a chance I might manage to do this on the train tomorrow.